### PR TITLE
New Feature Redux link with Views and API conecctions 

### PR DESCRIPTION
**Pull request by miguelesco**
## What's new 

- Use every API call from the documentation.
- Refactor all the code to add book and remove book features to persist your changes in the server.
- You should refactor the redux code of your application using middleware to make async requests to the API.
- Add the book and remove working in the same way but with Redux.

## Live Preview

https://miguelesco.github.io/bookstore/#/

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,31 +1,54 @@
...
-});
-
 const BookReducer = (state = initialState, action) => {
+  let booksList = [...state];
   switch (action.type) {
     case ADD_BOOK:
-      return [...state, action.payload];
+      booksList.push(action.payload);
+      return booksList;
+
```
Comment (dasileker) [inline src/redux/books/books.js:26]: - [x] The state to  store the new pushed book is not functioning, when I refresh the  page I loos the new add.

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @miguelesco,** 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 1 above)
Comment (miguelesco) [reply src/redux/books/books.js:26]: The live preview now is working was a piece of code I forget to delete

Comment (Emmyn5600) [reply src/redux/books/books.js:26]: Well solved

**Review by Emmyn5600 — APPROVED** [review]
Hi, 

Good job so far! 👏🏼 @miguelesco 

May I highlight the following:

- The repo looks professional.:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

## STATUS: APPROVED :muscle::clap:

Congratulations! :tada:
Your project is complete!  It's time to merge it :shipit:
Great work on implementing the changes requested by the previous reviewer.

#### Optional suggestion
 
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Pull request merged**